### ISSUE TITLE
How do you deal with deadlines? 

### ISSUE THREAD

**Issue by dman777** [notwaldorf/ama#87]
Deadlines are one of the things I hate most in software development. How do you, a professional software developer at Google, deal with deadlines at Google? 

(in the context of software development)
1) Have you ever missed a deadline at Google?
2) Do you work on weekends or off hours to meet deadlines?
3) Do you let stress or anxiety take over if an unexpected issue pops up possibly jeopardizing a deadline?

**Comment by notwaldorf** [comment]
Ha! This is a great question. I am pretty good at deadlines -- that's partially because I've got a bit of experience, and I know what I can do in a given amount of time. This often involves being very honest about what can and cannot be done in a given amount of time, and changing the schedule around it. It also helps that I'm a planner (in the sense that I have a stack of TODOs about every little upcoming thing, not that these TODOs are in any shape or form organized 😅) and I tend to remember what everyone around me is working on, so that I can make decisions based on that. This has helped me with estimating things in the past.

1. It's hard to say if I've ever "missed" a deadline --I'm sure that in the technical sense of the word I absolutely have. In both of the teams I've worked on the deliverables are fairly flexible -- "ship this on date X" can mean "ship most of the features" or "assuming no upstream blockers be ready to ship all the features" so it's very rare that you'll completely miss a deadline and not see it coming. Most of the time when I've "missed" a deliverable it's because of things that were out of my control, but I'm sure I've just gotten lucky.

2. I tend to be very against working evening or weekends on principle, but before big events (like Google I/O or a conference etc) where you *have* to have a specific thing done, I will go in crunch mode and work longer hours -- with the implied assumption that I will take days off once that event is done to make up for the late hours. 

3. I think "let" and "anxiety" don't really fit together as concepts -- anxiety happens whether or not you let it 🙃. That being said, I'm honestly not super great under pressure if I can't control the thing that's stressing me out. I'm ok if like....I have a bunch of work to do and I need to work more to do it, but I get pretty stressed out and grumpy when I had a plan and it fails, or if I depend on others doing a bunch of work and that ends up taking longer. It's not great and I usually regret feeling like that (especially if I'm shitty around coworkers) so I try to just deal with it. And maybe apologize afterwards and make sure everyone knows their work is valued (coworkers reading this: i love you alllllll 💕).

TL DR the hardest problem in computer science is people and time ❤️
